geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -209,9 +209,10 @@ object FilterHelper extends LazyLogging {
 
       // Note: although not technically required, all known spatial predicates are also binary spatial operators
       case f: BinarySpatialOperator if isSpatialFilter(f) =>
+        if (attribute == null) logger.warn("Geometry attribute is not set.  Will include all geometries from filter")
```

Comment:
actually you just want the envelope right? another thing you could do is to call `BoundsFilterVisitor(filter)` - that might be the best solution.